pkg/kv/kvserver/kvserverpb/state.proto

```diff
@@ -138,6 +138,12 @@ message ReplicaState {
   // RangeAppliedState.RaftAppliedIndexTerm.
   uint64 raft_applied_index_term = 14;
 
+  // gc_range_hint contains GC hint information for the replica. If hint
+  // timestamp is set that means all data in the range is expected to be
+  // garbage at that time and MVCC GC should optimize its removal togther
+  // with other related ranges to reduce load on pebble.
+  roachpb.GCRangeHint gc_range_hint = 15 [(gogoproto.customname) = "GCRangeHint"];
```

Comment:
Since the key and state methods are called `GCHint`, let's call this field and type `GCHint` as well. It's more general.